### PR TITLE
Redmine 25021 add phone field in transversal search functions

### PR DESCRIPTION
**Pull request by atos-mohamed-b**
This pull request is related to the following PR for the [_plugin-appointment_](https://github.com/lutece-secteur-public/gru-plugin-appointment):

- https://github.com/lutece-secteur-public/gru-plugin-appointment/pull/90

It depends on its resource files for the values of the new '_phone number_' fields.

### CODE REVIEW

File: src/java/fr/paris/lutece/plugins/appointment/modules/management/service/indexer/LuteceAppointmentSearchIndexer.java [hunk 1]
```diff
@@ -265,6 +265,15 @@ private Document getDocument( AppointmentDTO appointmentDTO, State appointmentSt
 
         doc.add( new StringField( AppointmentSearchItem.FIELD_MAIL_SEARCH, appointmentDTO.getEmail( ).toLowerCase( ), Field.Store.YES ) );
         doc.add( new SortedDocValuesField( AppointmentSearchItem.FIELD_MAIL_SEARCH, new BytesRef( appointmentDTO.getEmail( ).toLowerCase( ) ) ) );
+        
+        // --- Phone Number
+        // A User's phone number can be null in DB, so we make sure this variable is always initialized with a value
+        String phoneNumber = appointmentDTO.getPhoneNumber( ) != null ? appointmentDTO.getPhoneNumber( ) : "";
+        doc.add( new StringField( AppointmentSearchItem.FIELD_PHONE_NUMBER, phoneNumber, Field.Store.YES ) );
+        doc.add( new SortedDocValuesField( AppointmentSearchItem.FIELD_PHONE_NUMBER, new BytesRef( phoneNumber ) ) );
+
```
Comment (ryahiaoui) [inline src/java/fr/paris/lutece/plugins/appointment/modules/management/service/indexer/LuteceAppointmentSearchIndexer.java:274]: why you indexing the phone number twice with two different fields!!!

**Review by ryahiaoui — COMMENTED** [review]
why you indexing the phone number twice with two different fields!!!

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,8 @@
+/target/
+/.classpath
+/.project
+/.settings
+/nbactions.xml
+/nb-configuration.xml
+/bin/
+webapp/WEB-INF/conf/db.properties
```
Comment (seboo) [inline .gitignore]: This is not a standard .gitignore file, 
(it  is not necessary to provide a .gitignore file)

**Review by ryahiaoui — COMMENTED** [review]
why you indexing the phone number twice with two different fields!!!

**Comment by atos-mohamed-b** [comment]
A cleaner PR has been made for this feature:

- https://github.com/lutece-secteur-public/gru-module-appointment-management/pull/6

**Pull request closed without merging**